aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -160,6 +188,14 @@ module aptos_framework::object {
         create_object_internal(creator_address, obj_addr, false)
     }
 
+    /// Create a new object whose address is derived based on the creator account address and another object.
+    /// Derivde objects, similar to named objects, cannot be deleted.
+    public fun create_derived_object(creator: &signer, derive_ref: &DeriveRef): ConstructorRef {
```

Comment:
should this be user derived?

Reply:
It's derived from the user addr + object addr. Do you mean something else?

Reply:
create_user_derived_object

Reply:
I'd make this friend only for now.

Reply:
@davidiw , do we want users to call this to generate un-deletable objects?

Reply:
They can't be deleted